### ISSUE TITLE
ImportError: cannot import name 'HookimplOpts' from 'pluggy'

### ISSUE THREAD

**Issue by maycuatroi** [pytest-dev/pytest#12144]
## Error:
- Test files: [Github/Test Folder)[https://github.com/maycuatroi/drawio/tree/cb4f8ef6ce1f173e955a961963b1b22d530c67ee/tests]
```terminal
(base) binhna@DESKTOP-D9VLB13:/mnt/d/Github/drawio$ pytest
Traceback (most recent call last):
  File "/home/binhna/miniconda3/bin/pytest", line 5, in <module>
    from pytest import console_main
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/pytest/__init__.py", line 6, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/_pytest/assertion/__init__.py", line 10, in <module>
    from _pytest.assertion import rewrite
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/_pytest/assertion/rewrite.py", line 36, in <module>
    from _pytest.assertion import util
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/_pytest/assertion/util.py", line 23, in <module>
    from _pytest.config import Config
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/_pytest/config/__init__.py", line 43, in <module>
    from pluggy import HookimplOpts
ImportError: cannot import name 'HookimplOpts' from 'pluggy' (/home/binhna/miniconda3/lib/python3.11/site-packages/pluggy/__init__.py)
```

## Pip list
<details close>
  <summary>Pip list</summary>

```terminal
Package                                  Version
---------------------------------------- ---------------
a2wsgi                                   1.10.2
about-time                               4.2.1
absl-py                                  2.1.0
acme                                     2.9.0
aiofiles                                 23.2.1
aiohttp                                  3.9.3
aiosignal                                1.3.1
alembic                                  1.13.1
alive-progress                           3.1.5
altair                                   5.2.0
amqp                                     5.2.0
annotated-types                          0.6.0
anyio                                    4.3.0
apache-airflow                           2.6.3
apache-airflow-providers-amazon          8.18.0
apache-airflow-providers-apache-beam     5.6.1
apache-airflow-providers-cncf-kubernetes 8.0.0
apache-airflow-providers-common-io       1.3.0
apache-airflow-providers-common-sql      1.11.0
apache-airflow-providers-ftp             3.7.0
apache-airflow-providers-google          10.15.0
apache-airflow-providers-http            4.9.1
apache-airflow-providers-imap            3.5.0
apache-airflow-providers-postgres        5.10.1
apache-airflow-providers-sendgrid        3.4.0
apache-airflow-providers-sqlite          3.7.1
apache-airflow-providers-ssh             3.10.1
apache-beam                              2.54.0
apispec                                  5.2.2
appdirs                                  1.4.4
archspec                                 0.2.3
argcomplete                              3.2.2
argon2-cffi                              23.1.0
argon2-cffi-bindings                     21.2.0
arrow                                    1.3.0
asgiref                                  3.7.2
asn1crypto                               1.5.1
asttokens                                2.4.1
astunparse                               1.6.3
async-lru                                2.0.4
attrs                                    23.2.0
Babel                                    2.14.0
backoff                                  2.2.1
bcrypt                                   4.1.2
beautifulsoup4                           4.12.3
billiard                                 4.2.0
black                                    24.2.0
bleach                                   6.1.0
blinker                                  1.7.0
boltons                                  23.0.0
botasaurus                               3.2.24
botasaurus-proxy-authentication          1.0.11
boto3                                    1.34.47
botocore                                 1.34.47
Brotli                                   1.0.9
CacheControl                             0.14.0
cachelib                                 0.9.0
cachetools                               5.3.2
capsolver-extension-python               1.0.1
cattrs                                   23.2.3
celery                                   5.3.6
certifi                                  2024.2.2
cffi                                     1.16.0
channels                                 4.0.0
chardet                                  5.2.0
charset-normalizer                       3.3.2
chrome-extension-python                  1.0.1
chromedriver-autoinstaller-fix           1.0.5
click                                    8.1.7
click-didyoumean                         0.3.0
click-plugins                            1.1.1
click-repl                               0.3.0
clickclick                               20.10.2
cloudpickle                              2.2.1
cloudscraper                             1.2.71
colorama                                 0.4.6
coloredlogs                              15.0.1
colorlog                                 4.8.0
colour                                   0.1.5
comm                                     0.2.1
conda                                    24.3.0
conda-content-trust                      0.2.0
conda-libmamba-solver                    23.12.0
conda-package-handling                   2.2.0
conda_package_streaming                  0.9.0
ConfigUpdater                            3.2
connexion                                3.0.6
contourpy                                1.2.0
crcmod                                   1.7
cron-descriptor                          1.4.3
croniter                                 2.0.2
cryptography                             42.0.5
cycler                                   0.12.1
dataclasses-json                         0.6.4
db-dtypes                                1.2.0
debugpy                                  1.8.1
decorator                                5.1.1
defusedxml                               0.7.1
Deprecated                               1.2.14
diff_cover                               8.0.3
dill                                     0.3.8
distro                                   1.8.0
Django                                   5.0.2
django-ca                                1.27.0
django-celery-results                    2.5.1
django-cors-headers                      4.3.1
django-environ                           0.11.2
django-filter                            23.5
django-firebase-auth                     1.2.1
django-letsencrypt                       4.1.0
django-object-actions                    4.2.0
django-redis                             5.4.0
django-storages                          1.14.2
djangorestframework                      3.14.0
dm-tree                                  0.1.8
dnspython                                2.6.1
docopt                                   0.6.2
docutils                                 0.20.1
drf-excel                                2.4.0
drf-nested-routers                       0.93.5
drf-yasg                                 1.21.7
dtreeviz                                 2.2.2
email-validator                          1.3.1
et-xmlfile                               1.1.0
executing                                2.0.1
fastapi                                  0.110.0
fastavro                                 1.9.4
fasteners                                0.19
fastjsonschema                           2.19.1
ffmpy                                    0.3.2
filelock                                 3.13.1
firebase-admin                           6.5.0
flake8                                   7.0.0
Flask                                    2.2.5
Flask-AppBuilder                         4.3.1
Flask-Babel                              2.0.0
Flask-Caching                            2.1.0
Flask-JWT-Extended                       4.6.0
Flask-Limiter                            3.5.1
Flask-Login                              0.6.3
Flask-Session                            0.6.0
Flask-SQLAlchemy                         2.5.1
Flask-WTF                                1.2.1
flatbuffers                              24.3.7
fonttools                                4.49.0
fqdn                                     1.5.1
frozenlist                               1.4.1
fsspec                                   2024.2.0
gast                                     0.5.4
gcloud-aio-auth                          4.2.3
gcloud-aio-bigquery                      7.1.0
gcloud-aio-storage                       9.2.0
gcsfs                                    2024.2.0
gitdb                                    4.0.11
GitPython                                3.1.42
google-ads                               23.0.0
google-analytics-admin                   0.22.5
google-api-core                          2.17.1
google-api-python-client                 2.118.0
google-auth                              2.28.1
google-auth-httplib2                     0.2.0
google-auth-oauthlib                     1.2.0
google-cloud-aiplatform                  1.42.1
google-cloud-appengine-logging           1.4.1
google-cloud-audit-log                   0.2.5
google-cloud-automl                      2.13.1
google-cloud-batch                       0.17.11
google-cloud-bigquery                    3.17.2
google-cloud-bigquery-datatransfer       3.14.1
google-cloud-bigquery-storage            2.24.0
google-cloud-bigtable                    2.23.0
google-cloud-build                       3.23.1
google-cloud-compute                     1.16.1
google-cloud-container                   2.40.0
google-cloud-core                        2.4.1
google-cloud-datacatalog                 3.18.1
google-cloud-dataflow-client             0.8.8
google-cloud-dataform                    0.5.7
google-cloud-dataplex                    1.12.1
google-cloud-dataproc                    5.9.1
google-cloud-dataproc-metastore          1.15.1
google-cloud-dlp                         3.15.1
google-cloud-firestore                   2.15.0
google-cloud-kms                         2.21.1
google-cloud-language                    2.13.1
google-cloud-logging                     3.9.0
google-cloud-memcache                    1.9.1
google-cloud-monitoring                  2.19.1
google-cloud-orchestration-airflow       1.11.1
google-cloud-os-login                    2.14.1
google-cloud-pubsub                      2.19.4
google-cloud-redis                       2.15.1
google-cloud-resource-manager            1.12.1
google-cloud-run                         0.10.3
google-cloud-secret-manager              2.18.1
google-cloud-spanner                     3.42.0
google-cloud-speech                      2.24.1
google-cloud-storage                     2.14.0
google-cloud-storage-transfer            1.11.1
google-cloud-tasks                       2.16.1
google-cloud-texttospeech                2.16.1
google-cloud-translate                   3.15.1
google-cloud-videointelligence           2.13.1
google-cloud-vision                      3.7.0
google-cloud-workflows                   1.14.1
google-crc32c                            1.5.0
google-pasta                             0.2.0
google-re2                               1.1
google-resumable-media                   2.7.0
googleapis-common-protos                 1.62.0
gradio                                   4.21.0
gradio_client                            0.12.0
grapheme                                 0.6.0
graphviz                                 0.20.1
greenlet                                 3.0.3
grpc-google-iam-v1                       0.13.0
grpc-interceptor                         0.15.4
grpcio                                   1.62.1
grpcio-gcp                               0.2.2
grpcio-status                            1.60.1
gunicorn                                 21.2.0
h11                                      0.14.0
h5py                                     3.10.0
hdfs                                     2.7.3
httpcore                                 1.0.4
httplib2                                 0.22.0
httpx                                    0.27.0
huggingface-hub                          0.21.4
humanfriendly                            10.0
idna                                     3.6
importlib-metadata                       6.11.0
importlib_resources                      6.1.2
inflection                               0.5.1
iniconfig                                2.0.0
ipykernel                                6.29.3
ipython                                  8.22.2
ipywidgets                               8.1.2
isoduration                              20.11.0
isort                                    5.13.2
itsdangerous                             2.1.2
javascript-fixes                         1.1.14
jedi                                     0.19.1
jeepney                                  0.8.0
Jinja2                                   3.1.3
jmespath                                 1.0.1
joblib                                   1.3.2
josepy                                   1.14.0
Js2Py                                    0.74
json-merge-patch                         0.2
json-tricks                              3.17.3
json5                                    0.9.22
jsonpatch                                1.32
jsonpath-ng                              1.6.1
jsonpickle                               3.0.3
jsonpointer                              2.1
jsonschema                               4.21.1
jsonschema-specifications                2023.12.1
jupyter                                  1.0.0
jupyter_client                           8.6.0
jupyter-console                          6.6.3
jupyter_core                             5.7.1
jupyter-events                           0.9.0
jupyter-lsp                              2.2.4
jupyter_server                           2.13.0
jupyter_server_terminals                 0.5.2
jupyterlab                               4.1.4
jupyterlab_pygments                      0.3.0
jupyterlab_server                        2.25.4
jupyterlab_widgets                       3.0.10
keras                                    3.0.5
keyring                                  10.1
kiwisolver                               1.4.5
kombu                                    5.3.5
kubernetes                               29.0.0
kubernetes_asyncio                       29.0.0
langchain                                0.1.12
langchain-community                      0.0.28
langchain-core                           0.1.31
langchain-text-splitters                 0.0.1
langsmith                                0.1.25
lazy-object-proxy                        1.10.0
libclang                                 18.1.1
libmambapy                               1.5.3
limits                                   3.9.0
linkify-it-py                            2.0.3
lockfile                                 0.12.2
looker-sdk                               24.0.0
lxml                                     5.1.0
Mako                                     1.3.2
Markdown                                 3.6
markdown-it-py                           3.0.0
MarkupSafe                               2.1.5
marshmallow                              3.21.1
marshmallow-enum                         1.5.1
marshmallow-oneofschema                  3.1.1
marshmallow-sqlalchemy                   0.26.1
matplotlib                               3.8.3
matplotlib-inline                        0.1.6
mccabe                                   0.7.0
mdit-py-plugins                          0.4.0
mdurl                                    0.1.2
menuinst                                 2.0.1
mistune                                  3.0.2
ml-dtypes                                0.3.2
more-itertools                           10.2.0
mpld3                                    0.5.10
mpmath                                   1.3.0
msgpack                                  1.0.7
multidict                                6.0.5
mypy                                     1.9.0
mypy-extensions                          1.0.0
namex                                    0.0.7
nbclient                                 0.9.0
nbconvert                                7.16.2
nbformat                                 5.9.2
nest-asyncio                             1.6.0
networkx                                 3.2.1
notebook                                 7.1.1
notebook_shim                            0.2.4
numpy                                    1.26.4
nvidia-cublas-cu12                       12.1.3.1
nvidia-cuda-cupti-cu12                   12.1.105
nvidia-cuda-nvcc-cu12                    12.3.107
nvidia-cuda-nvrtc-cu12                   12.1.105
nvidia-cuda-runtime-cu12                 12.1.105
nvidia-cudnn-cu12                        8.9.2.26
nvidia-cufft-cu12                        11.0.2.54
nvidia-curand-cu12                       10.3.2.106
nvidia-cusolver-cu12                     11.4.5.107
nvidia-cusparse-cu12                     12.1.0.106
nvidia-nccl-cu12                         2.19.3
nvidia-nvjitlink-cu12                    12.3.101
nvidia-nvtx-cu12                         12.1.105
oauthlib                                 3.2.2
objsize                                  0.7.0
openai                                   1.14.1
opencv-python                            4.9.0.80
openpyxl                                 3.1.2
opentelemetry-api                        1.22.0
opentelemetry-exporter-otlp              1.22.0
opentelemetry-exporter-otlp-proto-common 1.22.0
opentelemetry-exporter-otlp-proto-grpc   1.22.0
opentelemetry-exporter-otlp-proto-http   1.22.0
opentelemetry-proto                      1.22.0
opentelemetry-sdk                        1.22.0
opentelemetry-semantic-conventions       0.43b0
opt-einsum                               3.3.0
ordered-set                              4.1.0
orjson                                   3.9.14
outcome                                  1.3.0.post0
overrides                                7.7.0
packaging                                23.1
pandas                                   2.2.0
pandas-gbq                               0.21.0
pandocfilters                            1.5.1
paramiko                                 3.4.0
parso                                    0.8.3
pathspec                                 0.9.0
pdfminer.six                             20231228
pdfplumber                               0.11.0
pendulum                                 2.0.0
pexpect                                  4.9.0
pillow                                   10.2.0
pip                                      24.0
platformdirs                             3.10.0
pluggy                                   1.4.0
ply                                      3.11
prison                                   0.2.1
prometheus_client                        0.20.0
prompt-toolkit                           3.0.43
proto-plus                               1.23.0
protobuf                                 4.25.3
psutil                                   5.9.8
psycopg2-binary                          2.9.9
ptyprocess                               0.7.0
pure-eval                                0.2.2
pyarrow                                  14.0.2
pyarrow-hotfix                           0.6
pyasn1                                   0.5.1
pyasn1-modules                           0.3.0
PyAthena                                 3.3.0
pycodestyle                              2.11.1
pycosat                                  0.6.6
pycparser                                2.21
pydantic                                 2.6.4
pydantic_core                            2.16.3
pydata-google-auth                       1.8.2
pydot                                    1.4.2
pydub                                    0.25.1
pyflakes                                 3.2.0
PyGithub                                 2.2.0
Pygments                                 2.17.2
pyjsparser                               2.7.1
PyJWT                                    2.8.0
pymongo                                  4.6.2
PyNaCl                                   1.5.0
pyOpenSSL                                23.2.0
pyparsing                                3.1.1
pypdfium2                                4.28.0
pyperclip                                1.8.2
pyRFC3339                                1.1
PySocks                                  1.7.1
pytest                                   8.1.1
pytest-metadata                          3.1.1
python-daemon                            3.0.1
python-dateutil                          2.9.0.post0
python-dotenv                            1.0.1
python-git                               2018.2.1
python-http-client                       3.3.7
python-json-logger                       2.0.7
python-multipart                         0.0.9
python-nvd3                              0.15.0
python-slugify                           8.0.4
pytz                                     2024.1
pytzdata                                 2020.1
PyYAML                                   6.0.1
pyzmq                                    25.1.2
qtconsole                                5.5.1
QtPy                                     2.4.1
redis                                    5.0.1
redshift-connector                       2.1.0
referencing                              0.33.0
regex                                    2023.12.25
requests                                 2.31.0
requests-oauthlib                        1.3.1
requests-toolbelt                        1.0.0
rfc3339-validator                        0.1.4
rfc3986-validator                        0.1.1
rich                                     13.7.1
rich-argparse                            1.4.0
rich-click                               1.7.4
rpds-py                                  0.18.0
rsa                                      4.9
ruamel.yaml                              0.17.21
ruff                                     0.3.2
s3transfer                               0.10.0
scikit-learn                             1.4.1.post1
scipy                                    1.12.0
scramp                                   1.4.4
seaborn                                  0.13.2
SecretStorage                            3.3.3
selenium                                 4.5.0
semantic-version                         2.10.0
semver                                   3.0.2
Send2Trash                               1.8.2
sendgrid                                 6.11.0
sentry-sdk                               1.42.0
setproctitle                             1.3.3
setuptools                               68.2.2
shapely                                  2.0.3
shellingham                              1.5.4
shortuuid                                1.0.11
six                                      1.16.0
smmap                                    5.0.1
sniffio                                  1.3.1
sortedcontainers                         2.4.0
soupsieve                                2.5
SQLAlchemy                               1.4.52
sqlalchemy-bigquery                      1.9.0
SQLAlchemy-JSONField                     1.0.2
sqlalchemy-redshift                      0.8.14
sqlalchemy-spanner                       1.6.2
SQLAlchemy-Utils                         0.41.1
sqlfluff                                 2.3.5
sqlparse                                 0.4.4
sshtunnel                                0.4.0
stack-data                               0.6.3
starkbank-ecdsa                          2.2.0
starlette                                0.36.3
sympy                                    1.12
tabulate                                 0.9.0
tblib                                    3.0.0
tenacity                                 8.2.3
tensorboard                              2.16.2
tensorboard-data-server                  0.7.2
tensorflow                               2.16.1
tensorflow-io-gcs-filesystem             0.36.0
termcolor                                2.4.0
terminado                                0.18.0
text-unidecode                           1.3
threadpoolctl                            3.3.0
time-machine                             2.14.0
tinycss2                                 1.2.1
tomlkit                                  0.12.0
toolz                                    0.12.1
torch                                    2.2.1
torchaudio                               2.2.1
torchinfo                                1.8.0
torchview                                0.2.6
torchvision                              0.17.1
tornado                                  6.4
tqdm                                     4.65.0
traitlets                                5.14.1
trio                                     0.24.0
trio-websocket                           0.11.1
triton                                   2.2.0
truststore                               0.8.0
typer                                    0.9.0
types-python-dateutil                    2.8.19.20240311
typing_extensions                        4.10.0
typing-inspect                           0.9.0
tzdata                                   2024.1
tzlocal                                  5.2
uc-micro-py                              1.0.3
unicodecsv                               0.14.1
universal_pathlib                        0.2.1
uri-template                             1.3.0
uritemplate                              4.1.1
urllib3                                  2.2.1
uvicorn                                  0.28.0
vine                                     5.1.0
watchtower                               3.0.1
wcwidth                                  0.2.13
webcolors                                1.13
webdriver-manager                        4.0.1
webencodings                             0.5.1
websocket-client                         1.7.0
websockets                               11.0.3
Werkzeug                                 3.0.1
wheel                                    0.43.0
widgetsnbextension                       4.0.10
wrapt                                    1.16.0
wsproto                                  1.2.0
WTForms                                  3.1.2
xgboost                                  2.0.3
yarl                                     1.9.4
zipp                                     3.17.0
zstandard                                0.19.0
```

</details>

- [x] a detailed description of the bug or problem you are having
- [x] output of `pip list` from the virtual environment you are using
- [x] pytest and operating system versions
    - OS: Ubuntu 22.04 in WSL2
    - Pytest: 8.1.1
- [x] minimal example if possible

**Comment by RonnyPfannschmidt** [comment]
this looks like a broken install issue - please ensure a compatible version of pluggy is installed

**Comment by maycuatroi** [comment]
I'm trying to install pluggy version 1.4.0 but it doesn't work



```terminal
(base) binhna@DESKTOP-D9VLB13:/mnt/d/Github/drawio$ pip install pluggy -U
Requirement already satisfied: pluggy in /home/binhna/miniconda3/lib/python3.11/site-packages (1.4.0)
(base) binhna@DESKTOP-D9VLB13:/mnt/d/Github/drawio$ pytest .
Traceback (most recent call last):
  File "/home/binhna/miniconda3/bin/pytest", line 5, in <module>
    from pytest import console_main
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/pytest/__init__.py", line 6, in <module>
    from _pytest.assertion import register_assert_rewrite
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/_pytest/assertion/__init__.py", line 10, in <module>
    from _pytest.assertion import rewrite
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/_pytest/assertion/rewrite.py", line 36, in <module> 
    from _pytest.assertion import util
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/_pytest/assertion/util.py", line 23, in <module>
    from _pytest.config import Config
  File "/home/binhna/miniconda3/lib/python3.11/site-packages/_pytest/config/__init__.py", line 43, in <module>
    from pluggy import HookimplOpts
ImportError: cannot import name 'HookimplOpts' from 'pluggy' (/home/binhna/miniconda3/lib/python3.11/site-packages/pluggy/__init__.py)
(base) binhna@DESKTOP-D9VLB13:/mnt/d/Github/drawio$
```

**Comment by RonnyPfannschmidt** [comment]
[the pluggy package at version 1.4.0](https://github.com/pytest-dev/pluggy/blob/1.4.0/src/pluggy/__init__.py) clearly has it - please show the content of your module and more details

this looks like a broken installation

**Comment by maycuatroi** [comment]
@RonnyPfannschmidt  Thank you so much. The problem is:

Somehow, pluggy version 1.0.0 was already installed in the site-packages. When I tried to install it via pip, although I attempted to uninstall and then reinstall it, the command output clearly indicated that version 1.4.0 had been installed. However, in reality, it was still using version 1.0.0 until I manually deleted the pluggy folder in the site packages. After reinstalling, it worked.
